pkg/api/defaults-kubelet.go

```diff
@@ -33,7 +33,7 @@ func (cs *ContainerService) setKubeletConfig(isUpgrade bool) {
 		switch key {
 		case "--anonymous-auth", "--client-ca-file":
 			if !to.Bool(o.KubernetesConfig.EnableSecureKubelet) { // Don't add if EnableSecureKubelet is disabled
-				staticLinuxKubeletConfig[key] = ""
+				delete(staticLinuxKubeletConfig, key)
```

Comment:
Now that we are no longer deleting empty string keys in the `removeKubeletFlags` func we do it here.